Apply the default operator on analyzed wildcard in simple_query_string builder

jimferenczi opened the pull request:
This is a followup from https://github.com/elastic/elasticsearch/pull/17711 where we now apply the default operator on analyzed wildcard query in query_string builder.

jpountz commented on core/src/main/java/org/elasticsearch/index/query/SimpleQueryParser.java:
```diff
@@ -167,62 +173,78 @@ private static Query wrapWithBoost(Query query, float boost) {
+                    tlist.add(currentPos);
                 }
+            } catch (IOException e) {
+                // ignore
```
!!!

jpountz replied:
don't try to fix it, you just moved code around, but this catch block worries me :(

jimferenczi replied:
Yep I was not sure of it too. This seems related to the fact that the simple query string should never throw exception. I'll add a comment to make sure that we revisit this later.

jpountz replied:
+1

jpountz commented on core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java:
```diff
@@ -407,4 +408,33 @@ public void testIndexMetaField() throws IOException {
+                }
+            }
+        }
+    }
```
we should also have a test with synonyms

jimferenczi replied:
Yep, I was looking for a test with a custom analyzer and could not find any. I think we should add a way to test custom analyzer directly in AbstractQueryTestCase. @cbuescher what do you think ?

cbuescher replied:
That depends a bit on how much it complicates the setup of AbstractQueryTestCase. I find that setup a bit too large and complex already. If there is an easy way to add a custom analyzer to the current setup I think it would be okay.

jpountz commented:
I left one comment about testing. Otherwise it looks great.

jimferenczi commented:
@jpountz thanks for the review. I've added tests with synonyms. 
@cbuescher I tried to add the ability to register custom analyzer in AbstractQueryTestCase but I could not find an easy way. Instead I've added a standalone test for SimpleQueryParser where I force the analyzer for the query.

jpountz commented:
LGTM
